core/controllers/reader_test.py

```diff
@@ -1104,58 +1104,98 @@ def test_get_recommendation_ids_with_invalid_author_recommended_ids(self):
+            json_response = self.put_json(
+                '/explorehandler/recommendations/%s' % self.EXP_ID_0, {
+                    'story_id': self.STORY_ID,
+                    'current_node_id': self.NODE_ID_2
+                }, csrf_token=csrf_token
+            )
+
+        self.assertEqual(json_response['summaries'][0]['id'], self.EXP_ID_1)
         self.logout()
 
-    def test_get_fails_when_story_does_not_exist_in_story_mode(self):
+    def test_put_fails_when_story_does_not_exist_in_story_mode(self):
+        self.login(self.NEW_USER_EMAIL)
+        csrf_token = self.get_new_csrf_token()
         with self.swap(constants, 'ENABLE_NEW_STRUCTURE_VIEWER_UPDATES', True):
-            self.get_json(
-                '/explorehandler/recommendations/%s' % self.EXP_ID_0, params={
+            self.put_json(
+                '/explorehandler/recommendations/%s' % self.EXP_ID_0, {
                     'story_id': 'invalid_story',
-                    'current_node_id': self.NODE_ID_2,
-                    'include_system_recommendations': False
-                }, expected_status_int=404
+                    'current_node_id': self.NODE_ID_2
+                }, csrf_token=csrf_token, expected_status_int=404
             )
+        self.logout()
 
-    def test_get_fails_when_node_does_not_exist_in_story_mode(self):
+    def test_put_fails_when_node_does_not_exist_in_story_mode(self):
+        self.login(self.NEW_USER_EMAIL)
+        csrf_token = self.get_new_csrf_token()
         with self.swap(constants, 'ENABLE_NEW_STRUCTURE_VIEWER_UPDATES', True):
-            self.get_json(
-                '/explorehandler/recommendations/%s' % self.EXP_ID_0, params={
+            self.put_json(
+                '/explorehandler/recommendations/%s' % self.EXP_ID_0, {
                     'story_id': self.STORY_ID,
-                    'current_node_id': 'invalid_node',
-                    'include_system_recommendations': False
-                }, expected_status_int=404
+                    'current_node_id': 'invalid_node'
+                }, csrf_token=csrf_token, expected_status_int=404
             )
+        self.logout()
+
 
-    def test_get_fails_when_story_is_not_published_in_story_mode(self):
+    def test_put_fails_when_story_is_not_published_in_story_mode(self):
         topic_services.unpublish_story(
             self.TOPIC_ID, self.STORY_ID, self.admin_id)
+        self.login(self.NEW_USER_EMAIL)
+        csrf_token = self.get_new_csrf_token()
+        with self.swap(constants, 'ENABLE_NEW_STRUCTURE_VIEWER_UPDATES', True):
+            self.put_json(
+                '/explorehandler/recommendations/%s' % self.EXP_ID_0, {
+                    'story_id': self.STORY_ID,
+                    'current_node_id': self.NODE_ID_2
+                }, csrf_token=csrf_token, expected_status_int=404
+            )
+        self.logout()
 
+    def test_put_returns_empty_string_when_user_completes_story(self):
```

Comment:
Not sure what you mean by "empty string" here. Do you mean "empty list"?

Reply:
Done, yeah.